source/AStar.hpp

```diff
@@ -33,7 +33,7 @@ namespace AStar
         uint getScore();
     };
 
-    using NodeSet = std::set<Node*>;
+    using NodeSet = std::vector<Node*>;
```

Comment:
Adding a node to NodeSet will be much faster (O(1)) than inserting in a ordered set (O(logN)). Furthermore you don't need to do comparisons.